src/GaugeInput.js

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Snackbar from '@material-ui/core/Snackbar';
+import Alert from '@material-ui/lab/Alert';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Button from '@material-ui/core/Button'
+
+const api = require('./api.js');
+
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    flexDirection: 'row',
+    width: '100%'
+  },
+  textField: {
+    width: '60%',
+    marginRight: 15
+  },
+  expansionPanelDetails: {
+    display: 'flex',
+    flexDirection: 'column'
+  }
+});
+
+class GaugeInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      id: `${this.props.id}`,
+      index: `${this.props.index}`,
+      prob: [],
```

Comment:
You could very well store `messages` in the state, equal to `this.props.gauge.messages`